main.cpp

```diff
@@ -32,399 +32,15 @@
-    ~ArpeggiatorWrapper()
-    {
-        delete pointerToArpeggiator;
-    }
-
-    Arpeggiator* pointerToArpeggiator = nullptr; //initialising member variable pointerToArpeggiator as nullptr
-};
-/*
- new UDT 4:
- with 2 member functions
- */
-struct MelodyGenerator
-{
-    // types above used as member variables
-    Sequencer sequencer;
-    Arpeggiator arpeggiator;
-    int  arpModeControlValue = arpeggiator.arpModeControlValue;
-    
-    MelodyGenerator();
-    ~MelodyGenerator();
-
-    void displayArpStatus();
-    void getSeqNoteRange();
-
-    JUCE_LEAK_DETECTOR(MelodyGenerator)
-};
-
-MelodyGenerator::MelodyGenerator()
-{
-    std::cout << "Hello from MelodyGenerator struct, I am being constructed" << std::endl;
-    std::cout << "MG arpmodevalue = " << arpModeControlValue << std::endl;    arpeggiator.arpModeControlValue = 0;
-}
-
-MelodyGenerator::~MelodyGenerator()
-{
-    arpeggiator.arpModeControlValue = 0;
-    std::cout << "Arp mode reset to default value of " << arpeggiator.arpModeControlValue << std::endl;
-    std::cout << "Goodbye from MelodyGenerator struct, I am being destructed" << std::endl;
-}
-// add 2 member functions that use member variables
-void MelodyGenerator::displayArpStatus()
-{
-       std::cout << "Melody generator arp mode = " << arpeggiator.getArpMode() << std::endl;
-}
-
-void MelodyGenerator::getSeqNoteRange()
-{
-    sequencer.displayNoteRange();
-}
-
-struct MelodyGeneratorWrapper
-{
-    MelodyGeneratorWrapper( MelodyGenerator* ptr ) : pointerToMelodyGenerator( ptr ) { } 
-    ~MelodyGeneratorWrapper()
-    {
-        delete pointerToMelodyGenerator;
-    }
-
-    MelodyGenerator* pointerToMelodyGenerator = nullptr; 
-};
-/*
- new UDT 5:
- with 2 member functions
- */
-struct SequencePlayer
-{
-    Oscillator oscillator;
-    Sequencer sequencer;
-
-    SequencePlayer();
-    ~SequencePlayer();
-
-    void displayOctaveAndRange();
-    void displayRootNote();
-
-    JUCE_LEAK_DETECTOR(SequencePlayer)
-};
-
-SequencePlayer::SequencePlayer()
-{
-    std::cout << "Hello from SequencePlayer struct, I am being constructed " << std::endl;
-}
-
-SequencePlayer::~SequencePlayer()
-{
-    std::cout << "Goodbye from SequencePlayer struct, I am being destructed " << std::endl;
-}
-//add 2 member functions
-void SequencePlayer::displayOctaveAndRange()
-{
-    oscillator.displayOctave();
-    sequencer.displayNoteRange();
-}
-
-void SequencePlayer::displayRootNote()
-{
-    std::cout << "Root note of sequence is " << oscillator.noteInHz << "hz" << std::endl;
-}
-
-struct SequencePlayerWrapper
-{
-    SequencePlayerWrapper( SequencePlayer* ptr ) : pointerToSequencePlayer( ptr ) { } 
-    ~SequencePlayerWrapper()
-    {
-        delete pointerToSequencePlayer;
-    }
-
-    SequencePlayer* pointerToSequencePlayer = nullptr; 
-};
-
+#include "Oscillator.h"
+#include "Sequencer.h"
+#include "Arpeggiator.h"
+#include "MelodyGenerator.h"
+#include "SequencePlayer.h"
+#include "Wrappers.h"
+#include "Wrappers.cpp"
```

Comment:
NEVER include cpp files.
including **literally** copies the entire contents of the file being included into the file with the `#include` statement.
Compile `wrappers.cpp` correctly, like you're compiling every other cpp file.
- [ ] revise